tests/Doctrine/Tests/Common/Proxy/ProxyGeneratorTest.php

```diff
@@ -229,6 +231,94 @@ public function testClassWithReturnTypesOnProxiedMethods()
         $this->assertEquals(1, substr_count($classCode, 'function returnsParent(): \stdClass'));
     }
 
+    public function testClassWithNullableTypeHintsOnProxiedMethods()
+    {
+        if (PHP_VERSION_ID < 70100) {
+            $this->markTestSkipped('Nullable type hints are only supported in PHP >= 7.1.0.');
+        }
+
+        if (!class_exists('Doctrine\Tests\Common\ProxyProxy\__CG__\NullableTypeHintsClass', false)) {
+            $className = NullableTypeHintsClass::class;
+            $metadata = $this->createClassMetadata($className, ['id']);
+
+            $proxyGenerator = new ProxyGenerator(__DIR__ . '/generated', __NAMESPACE__ . 'Proxy', true);
+            $this->generateAndRequire($proxyGenerator, $metadata);
+        }
+
+        $classCode = file_get_contents(__DIR__ . '/generated/__CG__DoctrineTestsCommonProxyNullableTypeHintsClass.php');
+
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintInt(?int $param)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintObject(?\stdClass $param)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintSelf(?\\' . $className . ' $param)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintWithDefault(?int $param = 123)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintWithDefaultNull(?int $param = NULL)'));
```

Comment:
Is it needed for 7.1 to generate 7.0 compatible code? If so, you could special case nullable parameters with a null default value to always omit the preceding `?`. It may not be exactly as written, but it will be compatible.